src/plugins/intel_cpu/src/nodes/subgraph.cpp

```diff
@@ -271,19 +286,41 @@ void Snippet::initSupportedPrimitiveDescriptors() {
         }
 
         impl_desc_type impl_type = impl_desc_type::unknown;
-        if (mayiuse(x64::avx512_core)) {
+#if defined(OPENVINO_ARCH_ARM64)
+        if (dnnl::impl::cpu::aarch64::mayiuse(dnnl::impl::cpu::aarch64::asimd)) {
+            impl_type = impl_desc_type::jit_asimd;
+        }
+#else
+        if (dnnl::impl::cpu::x64::mayiuse(x64::avx512_core)) {
             impl_type = impl_desc_type::jit_avx512;
-        } else if (mayiuse(x64::avx2)) {
+        } else if (dnnl::impl::cpu::x64::mayiuse(x64::avx2)) {
             impl_type = impl_desc_type::jit_avx2;
         }
+#endif
         return {config, impl_type};
     };
 
+    bool isPlanarPrioritized = false;
+#if defined(OPENVINO_ARCH_ARM64)
+    for (size_t i = 0; i < getParentEdges().size(); i++) {
+        const auto& parentNode = getParentEdgeAt(i)->getParent();
+        const auto& desc = parentNode->getSupportedPrimitiveDescriptors();
+        if (parentNode->getType() == Type::FullyConnected || parentNode->getType() == Type::MatMul) {
+            if (desc.size() == 1 && desc[0].getConfig().outConfs[0].getMemDesc()->hasLayoutType(ov::intel_cpu::LayoutType::ncsp)) {
+                isPlanarPrioritized = true;
+            }
+        }
+    }
+#endif
+
+    if (isPlanarPrioritized)
+        supportedPrimitiveDescriptors.emplace_back(initDesc(Planar));
```

Comment:
I think you can reuse [logic](https://github.com/openvinotoolkit/openvino/blob/2426259eaeed41feabc5d7445863c8aa98c5ecf1/src/plugins/intel_cpu/src/nodes/eltwise.cpp#L2518) from Eltwise. What do you think? It can be useful in your case?

Reply:
Thanks Alexandra for the comment! I tried that logic, it doesn't work. Because for this logic, nspc will be pushed to supportedPrimitiveDescriptors first as long as long ModelType is CNN. And a model is ModelType::CNN as long as it has Convolution( or ConvolutionBackpropData) node. The regressed model also has Convolution, so those subgraphs still select nspc, which is inefficient for this case. The issue here is the model has fullyconnected/matmul nodes that only supports ncsp on arm, we need to chose ncsp layout. Please feel free to have further discussions!